### ISSUE TITLE
[Enhancement] Calculer la moyenne d'une matière individuelle si celle-ci n'est pas renvoyée

### ISSUE THREAD

**Issue by BlueStarOne** [PapillonApp/papillon-v7#353]
### Description du bug

Mon lycée de retourne ni les moyennes générales ni les moyennes par matières sur pronote.
Dans papillon, je peux voir ma moyenne générale, mais pas mes moyennes par matières. Pourtant, je peux savoir de combien les moyennes par matières impactent ma moyenne générale. Papillon affiche donc -1/20 en tant que ma moyenne par matière, quelque soit ma matière 

### Étapes à reproduire

1. Avoir un lycée qui n'affiche pas les moyennes
2. Ouvrir papillon

### Comportement attendu

Afficher la moyenne par matière plus tôt que -1/20

### Appareil

OnePlus8

### Version du système d`exploitation

A14

### Version

7.3.0

### Environnement

Application native Android

### Source de l'application

Stores (Play Store / App Store)

### Service scolaire

🦋 Pronote

### Captures d'écran / vidéo

Mon tel ne veut pas me laisser ajouter, voici un lien temporaire vers l'image https://cdn.discordapp.com/attachments/1014932976636809316/1305971663980073011/Screenshot_20241112-200349_Papillon.png

**Comment by Kgeek33** [comment]
Si ton lycée bloque l'accès aux moyennes généralles, Papillon fera de même
Cependant, j'ai fait un fix (ou sur ma pr #325 ) pour remplacer -1/20 par ??/20

**Comment by Kgeek33** [comment]
ok laisse tomber mon msg précédent, j'avais pas compris. J'croyais que tu parlais sur les meilleures/pires notes

donc ça serait une amélioration donc ? pask Papillon fait la même chose que Pronote. Quand les moyennes sont bloquées, ça retourne -1/20

**Comment by BlueStarOne** [comment]
Oui, ça peut être compris comme une amélioration... Je savais pas exactement si c'était un bug ou juste qqch de normal

**Comment by Kgeek33** [comment]
on peut penser que c'est un bug, mais c'est en fait un comportement normal de Papillon

**Comment by BlueStarOne** [comment]
Pourtant, papillon calcule correctement ma moyenne générale. 
Pourquoi est-ce qu'il ne calcule pas aussi ma moyenne par matière ? 
Et si il ne la calcule pas, comment sait-il que telle matière enlève x points sur ma moyenne générale et que telle matière en rajoute ?

**Comment by Kgeek33** [comment]
Alors faudra que je regarde mais j'crois que ta moyenne générale est quand même retourné, elle est juste cachée sur Pronote, c'est pour ça que Papillon réussi à voir l'influence qu'à une note sur la moyenne

Par contre, ce dont je suis sûr, c'est que la moyenne de la classe est calculée, donc elle sera différente de Pronote

**Comment by LeGeek01** [comment]
non la moyenne générale n'est pas cachée, si l'etab l'a désactivé Papillon la calcule, ça j'en suis 100% sur et certain de par la différence de la moyenne retournée par Papillon et celle que mon prof a sur son tableau

**Comment by Kgeek33** [comment]
ah ben ce sont des points à améliorer alors sur Papillon

**Comment by ecnivtwelve** [comment]
Je renomme l'issue selon ce que j'en ai compris, corrige-moi si besoin

**Comment by ecnivtwelve** [comment]
Je note qu'il y a un problème similaire lorsqu'aucune note n'est coefficientée

![image](https://github.com/user-attachments/assets/f1649df2-8154-4b7c-945e-528febdc4079)
![image](https://github.com/user-attachments/assets/2590b01d-e5fb-4138-a386-963bb806790c)

**Comment by Kgeek33** [comment]
J'en profite pour réveiller l'issue pour signaler un problème similaire.

- J'ai une moyenne de 20 en Nsi
- J'ai une moyenne vide en Maths (car la note est coeff 0)

Comme une moyenne vide renvoie -1/20, lorsqu'on calcule l'influence de la moyenne en nsi sur la moyenne générale... (pas besoin de poursuivre)
![Screenshot_2024-12-17-00-24-04-503_xyz.getpapillon.app.jpg](https://github.com/user-attachments/assets/f61b7608-56e7-49e8-82b8-dc59c0c41023)
